mirage/resolver/dns_resolver_mirage.mli

```diff
@@ -1,8 +1,9 @@
 (* (c) 2017, 2018 Hannes Mehnert, all rights reserved *)
 
 module Make (S : Tcpip.Stack.V4V6) : sig
-
-  val resolver : S.t -> ?root:bool -> ?timer:int -> ?udp:bool -> ?tcp:bool -> ?tls:Tls.Config.server -> ?port:int -> ?tls_port:int -> Dns_resolver.t -> unit
+  val resolver
+    :  S.t -> ?root:bool -> ?timer:int -> ?udp:bool -> ?tcp:bool -> ?tls:Tls.Config.server -> ?port:int -> ?tls_port:int
+    -> Dns_resolver.t -> ((Ipaddr.t * int) -> string -> string Lwt.t)
```

Comment:
I think the return type deserves a comment!

Reply:
Perhaps the return type should be an opaque `type resolver` and we add a function `val resolve_external : (Ipaddr.t * int) -> string -> string Lwt.t` that just calls the function. Then we can more easily change the implementation.